File: Music.Core/Models/Music/Artist.cs
namespace Music.Core.Models.Music;

public class Artist : Music
{
    public string GetImageUrl(Image.QQMusic image)
        => $"https://y.qq.com/music/photo_new/T001R{ (int)image }x{ (int)image }M000{ Id }.jpg";

    private List<Song>? songs;
    public List<Song> Songs
    {
        get
        {
            if (songs is null)
            {
                songs = new();
                foreach (var song in Repository.Songs)
                {
                    foreach (var artist in song.Artists)
                    {
                        if (artist.Id is null) continue;
                        if (artist.Id.Equals(Id)) songs.Add(song); continue;
                    }
                }
            }
            return songs;
        }
    }
}

File: Music.Client.Android/Java/short/music/src/main/java/com/zitherharp/music/model/Artist.java
package com.zitherharp.music.model;

import com.zitherharp.music.core.QQMusic;

import org.jetbrains.annotations.NotNull;

import java.util.ArrayList;
import java.util.List;

public class Artist extends QQMusic {
    private List<Short> shorts;

    @Override
    public String getImageUrl(@NotNull Image image) {
        return String.format("https://y.qq.com/music/photo_new/T001R%sM000%s.jpg", image.getSize(), Id);
    }
}
